lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -287,8 +287,20 @@ static void printUnpackedTypePostfix(Type type, raw_ostream &os) {
 
 /// Return the word (e.g. "reg") in Verilog to declare the specified thing.
 static StringRef getVerilogDeclWord(Operation *op) {
-  if (isa<RegOp>(op))
+  if (isa<RegOp>(op)) {
+    // Check if the type stored in this register is a struct or array of
+    // structs. In this case, according to spec section 6.8, the "reg" prefix
+    // should be left off.
+    auto elementType =
+        op->getResult(0).getType().cast<InOutType>().getElementType();
+    if (elementType.isa<StructType>())
+      return "";
+    if (auto innerType = elementType.dyn_cast<ArrayType>())
```

Comment:
What happens on multidimensional arrays around a struct? Should this be a while loop unwrapping the arrays?

Reply:
Ah, good point, will add that and a test case.